Source/WebCore/page/Navigation.cpp

```diff
@@ -385,6 +385,14 @@ void Navigation::resolveFinishedPromise(NavigationAPIMethodTracker* apiMethodTra
     cleanupAPIMethodTracker(apiMethodTracker);
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#reject-the-finished-promise
+void Navigation::rejectFinishedPromise(NavigationAPIMethodTracker* apiMethodTracker, Exception&& exception, JSC::JSValue exceptionObject)
+{
+    apiMethodTracker->finishedPromise->reject(exception, RejectAsHandled::Yes, exceptionObject);
+    apiMethodTracker->committedPromise->reject(exception, RejectAsHandled::No, exceptionObject);
```

Comment:
Maybe needs a comment why for comitted reject does not need to be marked as handled.

Reply:
I wasn't certain about this. The specs are clear that finished is marked as handled earlier (this just overrides that), so I assumed the default was not to be.

Reply:
Should we assert at this point the finished promise is marked as handled and committed promise is not?

Reply:
I tested Chromium and it matches the behavior here.

I'll add a comment. DefferredPromise doesn't even have a way currently to query markedAsHandled and I'm not sure an assert is very valuable.